internal/daemon/vendor-specific-plugins/marvell/main.go

```diff
@@ -312,13 +468,12 @@ func (vsp *mrvlVspServer) Listen() (net.Listener, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to listen on the vendor plugin socket: %v", err)
 	}
-
+	mrvlVspServerInstance := &mrvlVspServer{}
```

Comment:
I think, it should not create a new instance here. Instead, the caller `main()` already has a `mrvlVspServer` variable. Let it pass it to `Listen()` as argument.